imagekitio/url.py

```diff
@@ -130,6 +130,13 @@ def get_signature_timestamp(expiry_seconds: int = None) -> int:
 
     @staticmethod
     def get_signature(private_key, url, url_endpoint, expiry_timestamp: int) -> str:
+        last_slash_pos = url.rfind('/')
+        question_mark_pos = url.find('?', last_slash_pos)
+        path = url[last_slash_pos + 1:question_mark_pos] if question_mark_pos != -1 else url[last_slash_pos + 1:]
+        encoded_path = Url.encode_string_if_required(path)
+        encoded_url = url[:last_slash_pos + 1] + encoded_path + url[question_mark_pos:] if question_mark_pos != -1 else url[:last_slash_pos + 1] + encoded_path
+        url = encoded_url
```

Comment:
This logic is wrong. Please refer this document on how to generate signed Url: https://docs.imagekit.io/features/security/signed-urls#generating-signed-urls

This will generate wrong signature for any asset not in root i.e. if url is 
`https://ik.imagekit.io/test_url_endpoint/folder,test/four-penguins-with-é.webp?q=rt` it will only encode `four-penguins-with-é.webp` and not `/folder,test/four-penguins-with-é.webp`.

Reply:
have updated the logic with relevant test cases